Sources/Layout/InverseLayout.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  InverseLayout.swift
+//  CollectionKit
+//
+//  Created by Luke Zhao on 2017-09-08.
+//  Copyright © 2017 lkzhao. All rights reserved.
+//
+
+import UIKit
+
+open class InverseLayout<Data>: WrapperLayout<Data> {
+
+  open override var contentSize: CGSize {
+    return rootLayout.contentSize.inverted
+  }
+
+  open override func layout(collectionSize: CGSize,
+                            dataProvider: CollectionDataProvider<Data>,
+                            sizeProvider: @escaping (Int, Data, CGSize) -> CGSize) {
+    rootLayout.layout(collectionSize: collectionSize.inverted,
+                      dataProvider: dataProvider) {
+                        return sizeProvider($0, $1, $2.inverted).inverted
```

Comment:
Does the size need to be inverted?

For example for a custom view its size is (width: 60, height: 80)

No matter for vertical or horizontal, I want it to have a size like (60, 80), right?

Reply:
yea. That’s what I did. It should work as expected

Reply:
I see, because you inverted it twice?

Reply:
The first invert is inverting the collectionSize to the original axis. Then we invert the size passed from the user.